File: src/main/java/ru/nsu/fit/team_project/model/CommandExecutor.java
package ru.nsu.fit.team_project.model;

public class CommandExecutor {
    private final Model model;

    public CommandExecutor(Model model) {
        this.model = model;
    }

    public void executeAll() {
        if (model.getCommands() != null) {
            model.getCommands().forEach(c -> c.execute(model));
        }
    }
}

File: src/jvmMain/kotlin/data/commands/CommandExecutor.kt
package data.commands

import data.commands.entity.Command
import data.model.ObjectsStorage

class CommandExecutor(
    private val storage: CommandsStorage,
) {

    fun executeAll(objectStorage: ObjectsStorage) {
        this.storage.commands.forEach {
            println("${it.id}")
            it.execute(objectStorage)
        }
    }

    fun executeList(commands: List<Command>, objectStorage: ObjectsStorage) {
        commands.forEach { it.execute(objectStorage) }
    }

    fun execute(command: Command, objectStorage: ObjectsStorage) {
        command.execute(objectStorage)
    }
}